shared_model/interfaces/common_objects/pager.hpp

```diff
@@ -40,9 +40,14 @@ namespace shared_model {
       /**
        * Max limit
        */
-      static constexpr Limit MAX_PAGER_LIMIT = 100;
-      static_assert(MAX_PAGER_LIMIT == iroha::model::Pager::MAX_PAGER_LIMIT,
-                    "Should be equal to old fashioned model's");
+      struct MaxLimit {
+        constexpr operator Limit() const {
```

Comment:
Why it isn't a static fn of Pager? Don't think there's need of this struct

Reply:
At first I use `constexpr static` but linker error happens (duplicate symbols).

Reply:
Next, I tried:
```
constexpr static uint16_t limit() { return 100; }
```
But it cannot guarantee `static_assert()` with old model because even constexpr function cannot be used in `static_assert()`

Reply:
`inline constexpr` should work

Reply:
I tried:
```cpp
      inline static constexpr maxLimit() { return 100; }
      static_assert(maxLimit() == iroha::model::Pager::MAX_PAGER_LIMIT,
                    "Should be equal to old fashioned model's");
```

GCC in docker shows:
```
/opt/iroha/shared_model/interfaces/common_objects/pager.hpp:43:40: error: ISO C++ forbids declaration of 'maxLimit' with no type [-fpermissive]
       inline static constexpr maxLimit() { return 100; }
                                        ^
/opt/iroha/shared_model/interfaces/common_objects/pager.hpp:44:7: error: non-constant condition for static assertion
       static_assert(maxLimit() == iroha::model::Pager::MAX_PAGER_LIMIT,
       ^
/opt/iroha/shared_model/interfaces/common_objects/pager.hpp:44:29: error: 'static constexpr int shared_model::interface::Pager::maxLimit()' called in a constant expression
       static_assert(maxLimit() == iroha::model::Pager::MAX_PAGER_LIMIT,
```

Reply:
I tried:
```cpp
      inline static constexpr uint16_t maxLimit() { return 100; }
      static_assert(maxLimit() == iroha::model::Pager::MAX_PAGER_LIMIT,
                    "Should be equal to old fashioned model's");
```

But unfortunately, gcc shows:
```
/opt/iroha/shared_model/interfaces/common_objects/pager.hpp:44:7: error: non-constant condition for static assertion
       static_assert(maxLimit() == iroha::model::Pager::MAX_PAGER_LIMIT,
       ^
/opt/iroha/shared_model/interfaces/common_objects/pager.hpp:44:29: error: 'static constexpr uint16_t shared_model::interface::Pager::maxLimit()' called in a constant expression
       static_assert(maxLimit() == iroha::model::Pager::MAX_PAGER_LIMIT,
                             ^
```

Reply:
Meh, `constexpr` implies `inline`, and there's no need to write `static`
Quite weird that it's not working, look at [this](https://ideone.com/dHbGCo)

Reply:
It seems that it doesn't work in class. https://wandbox.org/permlink/UB8KAVQPGdrkKjOz
I'll change it to global constexpr variable.
Following works correctly.
```cpp
namespace shared_model {
  namespace interface {
    struct Pager {...};
    constexpr Pager::Limit MAX_PAGER_LIMIT = 100;
    static_assert(MAX_PAGER_LIMIT == iroha::model::Pager::MAX_PAGER_LIMIT,
                  "Should be equal to old fashioned model's");
  }
}
```